File: 3-Big O/big_o/lib/find_nemo2.ex
defmodule FindNemo2 do
  def find(lst) do
    t0 = System.monotonic_time(:millisecond)

    for x <- 0..length(lst), Enum.at(lst, x) == "nemo" do
      IO.puts("found NEMO!")
    end

    t1 = System.monotonic_time(:millisecond)
    IO.puts("Call to find Nemo took #{t1 - t0} Milliseconds")
  end
end

# => O(n) Linear Time
# nemo = ["nemo"]
# everyone = ~w(dory bruce marlin nemo gill bloat nigel squirt darla hank)
# FindNemo2.find(everyone)
# FindNemo2.find(List.duplicate("nemo", 100000))

File: 3-Big_O/find_nemo2.js
const findNemo = arr => {
  let t0 = performance.now();

  for (let i = 0; i < arr.length; ++i) {
    if (arr[i] === 'nemo') {
      console.log('Found NEMO!');
    }
  }

  let t1 = performance.now();
  console.log('Call to find Nemo took ' + (t1 - t0) + ' Milliseconds');
};

const nemo = ['nemo'];
const everyone = [
  'dory',
  'bruce',
  'marlin',
  'nemo',
  'gill',
  'bloat',
  'nigel',
  'squirt',
  'darla',
  'hank',
];

const large = new Array(100000).fill('nemo');
findNemo(large); // O(n) Linear Time
